bin/gather_pipelines

```diff
@@ -48,22 +49,31 @@ echo "gather_pipelines:$LINENO] inspecting roles rolebindings in tekton-pipeline
 
 # Collect logs and yaml specs for pods referenced by TaskRuns
 NAMESPACES=$(oc get taskruns,eventlisteners --all-namespaces=true --ignore-not-found -o jsonpath='{range .items[*]}{@.metadata.namespace}{"\n"}{end}' | uniq)
-for NAMESPACE in ${NAMESPACES[@]}; do 
+for NAMESPACE in ${NAMESPACES[@]}; do
     # get all the pods names referenced by taskruns
     PODS=$(oc get po -n ${NAMESPACE} -o json | jq -r '.items[].metadata | select (.labels."app.kubernetes.io/managed-by"=="tekton-pipelines" or .labels."app.kubernetes.io/managed-by"=="EventListener")| .name' | tr  '\n' ' ')
     mkdir -p ${LOGS_DIR}/namespaces/${NAMESPACE}/core
     mkdir -p ${LOGS_DIR}/namespaces/${NAMESPACE}/pods
+    mkdir -p ${LOGS_DIR}/namespaces/${NAMESPACE}/events
+
     echo "gather_pipelines:$LINENO] collecting eventlisteners and taskrun pods in ${NAMESPACE} .."
     eval oc get pod ${PODS} -n ${NAMESPACE} -o=yaml >${LOGS_DIR}/namespaces/${NAMESPACE}/core/pods.yaml
-    echo "gather_pipelines:$LINENO] inspecting endpoints, services and route in ${NAMESPACE} .."
-    oc adm inspect -n ${NAMESPACE} $(oc get svc,ep,routes -o name -n ${NAMESPACE}) --dest-dir=${LOGS_DIR} > /dev/null
-    for POD in ${PODS[@]}; do 
+    output=$(oc get svc,ep,routes -n ${NAMESPACE} --ignore-not-found)
+    # Check if the output is empty (no objects found)
+    if [ -z "$output" ]; then
+        echo "The requested objects(svc,ep,routes) do not exist in ${NAMESPACE}"
+    else
+        echo "gather_pipelines:$LINENO] inspecting endpoints, services and route in ${NAMESPACE} .."
+        oc adm inspect -n ${NAMESPACE} $(oc get svc,ep,routes -o name -n ${NAMESPACE}) --dest-dir=${LOGS_DIR} > /dev/null
+    fi
+    for POD in ${PODS[@]}; do
         echo "gather_pipelines:$LINENO] collecting logs from pod ${POD} in ${NAMESPACE} .."
         mkdir -p ${LOGS_DIR}/namespaces/${NAMESPACE}/pods/${POD}
         CONTAINERS=$(oc get po -n ${NAMESPACE} ${POD} -o jsonpath='{range .spec.containers[*]}{@.name}{" "}{end}')
-        for CONTAINER in ${CONTAINERS[@]}; do 
+        for CONTAINER in ${CONTAINERS[@]}; do
             mkdir -p ${LOGS_DIR}/namespaces/${NAMESPACE}/pods/${POD}/${CONTAINER}/${CONTAINER}/logs
             oc logs -n ${NAMESPACE} ${POD} -c ${CONTAINER} &> ${LOGS_DIR}/namespaces/${NAMESPACE}/pods/${POD}/${CONTAINER}/${CONTAINER}/logs/current.log
+            oc get pods -n ${NAMESPACE} ${POD} -oyaml > ${LOGS_DIR}/namespaces/${NAMESPACE}/pods/${POD}/${POD}.yaml
```

Comment:
Aren't the pods resources yaml already collected [here](https://github.com/openshift-pipelines/must-gather/pull/38/commits/01a773e24a7a31b381b625c77e3712614f148856#diff-656aa18ee96c8c2444f69b7bccc55578df634cde1f88c037852df00724c962daR60)? It sounds like a repetition.

Reply:
One question i had is there is only one yaml for all pods 
is that fine ?

Reply:
Same for other tekton resources i.e pipelineruns, taskruns

Reply:
Yes, It is needed for namespaced resources to have them in a single yaml file (per namespace).

Reply:
Will having separate yaml also per resource break omc ?

Reply:
nope if you put them in `${LOGS_DIR}/namespaces/${NAMESPACE}/${API_GROUP}/${API_PLURAL_NAME}/`